Content.Client/_DeltaV/CartridgeLoader/Cartridges/NanoChatUiFragment.xaml.cs

```diff
@@ -0,0 +1,254 @@
+using System.Linq;
+using System.Numerics;
+using Content.Shared._DeltaV.CartridgeLoader.Cartridges;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Client.UserInterface;
+using Robust.Shared.Timing;
+
+namespace Content.Client._DeltaV.CartridgeLoader.Cartridges;
+
+[GenerateTypedNameReferences]
+public sealed partial class NanoChatUiFragment : BoxContainer
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    private const int MaxMessageLength = 256;
+
+    private readonly NewChatPopup _newChatPopup;
+    private uint? _currentChat;
+    private uint? _pendingChat;
+    private uint _ownNumber;
+    private bool _notificationsMuted;
+    private Dictionary<uint, NanoChatRecipient> _recipients = new();
+    private Dictionary<uint, List<NanoChatMessage>> _messages = new();
+
+    public event Action<NanoChatUiMessageType, uint?, string?, string?>? OnMessageSent;
+
+    public NanoChatUiFragment()
+    {
+        IoCManager.InjectDependencies(this);
+        RobustXamlLoader.Load(this);
+
+        _newChatPopup = new NewChatPopup();
+        SetupEventHandlers();
+    }
+
+    private void SetupEventHandlers()
+    {
+        _newChatPopup.OnChatCreated += (number, name, job) =>
+        {
+            OnMessageSent?.Invoke(NanoChatUiMessageType.NewChat, number, name, job);
+        };
+
+        NewChatButton.OnPressed += _ =>
+        {
+            _newChatPopup.ClearInputs();
+            _newChatPopup.OpenCentered();
+        };
+
+        MuteButton.OnPressed += _ =>
+        {
+            _notificationsMuted = !_notificationsMuted;
+            UpdateMuteButton();
+            OnMessageSent?.Invoke(NanoChatUiMessageType.ToggleMute, null, null, null);
+        };
+
+        MessageInput.OnTextChanged += args =>
+        {
+            var length = args.Text.Length;
+            var isValid = !string.IsNullOrWhiteSpace(args.Text) &&
+                          length <= MaxMessageLength &&
+                          (_currentChat != null || _pendingChat != null);
+
+            SendButton.Disabled = !isValid;
+
+            // Show character count when over limit
+            CharacterCount.Visible = length > MaxMessageLength;
+            if (length > MaxMessageLength)
+            {
+                CharacterCount.Text = Loc.GetString("nano-chat-message-too-long",
+                    ("current", length),
+                    ("max", MaxMessageLength));
+                CharacterCount.StyleClasses.Add("LabelDanger");
+            }
+        };
+
```

Comment:
```suggestion

MessageInput.OnTextEntered += _ => SendMessage(); // Send message when press enter
```

i add this to my downstream when a i port the messaging

Reply:
Okay I'll add this too

Reply:
resolved with [0aec325](https://github.com/Goob-Station/Goob-Station/pull/1269/commits/0aec3258bb1dded4a680f82373932a6c4bc820d3)